packages/router-generator/src/generator.ts

```diff
@@ -359,17 +359,17 @@ export const Route = createAPIFileRoute('${escapedRoutePath}')({
         await prettier.format(replaced, prettierOptions),
       )
     } else {
-      const copied = routeCode.replace(
-        /(createAPIFileRoute\(\s*['"])([^\s]*)(['"],?\s*\))/g,
-        (_, p1, __, p3) => `${p1}${escapedRoutePath}${p3}`,
+      const copied = await prettier.format(
```

Comment:
Oh, that's a neat catch. In hindsight, it seems so obvious that the `routeCode` would be formatted differently against what we are comparing it towards.

We should do this for the `routeCode` checked above as well as the final `routeTree` write.

Reply:
we definitely should bring this to main asap, independent from HMR

Reply:
I pulled those changes out of this branch and merged to main